src/TestData/Factories/DonationFormFactory.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace GiveTestData\TestData\Factories;
+
+use GiveTestData\TestData\Framework\Factory;
+
+/**
+ * Class DonationFormFactory
+ * @package GiveTestData\TestData\Factories
+ */
+class DonationFormFactory extends Factory {
+
+	/**
+	 * @var bool
+	 */
+	private $donationGoal;
+
+	/**
+	 * @var bool
+	 */
+	private $termsAndConditions;
+
+	/**
+	 * @var string
+	 */
+	private $template;
+
+	/**
+	 * @var string[]
+	 */
+	private $templates = [ 'sequoia', 'legacy' ];
+
+	/**
+	 * @param string $template
+	 *
+	 * @return bool
+	 */
+	public function checkFormTemplate( $template ) {
+		if ( 'random' === $template ) {
+			return true;
+		}
+
+		return in_array( $template, $this->templates );
+	}
+
+	/**
+	 * @param string $template
+	 */
+	public function setFormTemplate( $template ) {
+		$this->template = $template;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getFormTemplate() {
+		if ( 'random' === $this->template ) {
+			return $this->randomDonationTemplate();
+		}
+
+		return $this->template;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function randomDonationTemplate() {
+		return $this->faker->randomElement( $this->templates );
+	}
+
+	/**
+	 * @param bool $generate
+	 */
+	public function setDonationFormGoal( $generate ) {
+		$this->donationGoal = ( bool ) $generate;
+	}
+
+	/**
+	 * @return false|string
+	 */
+	public function getDonationGoal() {
+		if ( is_null( $this->donationGoal ) || ! $this->donationGoal ) {
+			return false;
+		}
+
+		return $this->randomGoal();
+	}
+
+	/**
+	 * @param bool $generate
+	 */
+	public function setTermsAndConditions( $generate ) {
```

Comment:
Thinking about it again, I think we can reduce the code footprint by delegating default values to the command class. So instead of calling a setter, the command would parse args and apply defaults on top of the returned data structure. This would separate the concerns of generation (the factory) and default values (the command).

Does that make sense?

Reply:
@kjohnson Instead of using Factory class setters and getters, you are suggesting to "redefine" the returned definition in the command class? Hmm... that might increase the code footprint, not the opposite.  Remember that we have two "entry points", CLI and REST API, and we will have to do that two times. Also, abstracting that work into a separate method/function is not much different than using setters and getters, but it's definitely more confusing. In my opinion, the Factory class should know how to make the "object" as this is the only responsibility and purpose of this class. But, maybe I got it all wrong :smile:  
Let me know what you think.